pandas/tests/arithmetic/test_datetime64.py

```diff
@@ -2350,3 +2352,34 @@ def test_shift_months(years, months):
            for x in dti]
     expected = DatetimeIndex(raw)
     tm.assert_index_equal(actual, expected)
+
+
+def test_add_with_monkeypatched_datetime(monkeypatch):
+    # GH 25734
+
+    class MetaDatetime(type):
+        @classmethod
+        def __instancecheck__(self, obj):
+            return isinstance(obj, datetime)
+
+    class FakeDatetime(MetaDatetime("NewBase", (datetime,), {})):
+        pass
```

Comment:
A metaclass is needed in this test case to be able to override __instancecheck__